src/wc.js

```diff
@@ -0,0 +1,159 @@
+/*  
+    copy tail.sh as an example, and provide a similar function to wc.
+    wc man page:http://man7.org/linux/man-pages/man1/wc.1.html
+    (notes from man page)        
+       wc - Print newline, word, and byte counts for each FILE, and a total line
+       if more than one FILE is specified.  A word is a non-zero-length
+       sequence of characters delimited by white space.
+        -c, --bytes print the byte counts
+        -m, --chars print the character counts
+        -l, --lines print the newline counts
+        --files0-from=F
+                    read input from the files specified by NUL-terminated names in
+                    file F; If F is - then read names from standard input
+        -L, --max-line-length print the maximum display width
+        -w, --words print the word counts
+        --help display this help and exit
+        --version output version information and exit
+*/
+
+var common = require('./common');
+var fs = require('fs');
+
+common.register('wc', _wc, {
+    canReceivePipe: true,
+    cmdOptions: {
+        'm': 'charCount',
+        'l': 'lineCount',
+        'w': 'wordCount',
+    },
+});
+
+function getLines(data) {
+    //data is the file contents
+    var lines = data.split('\n'); // array
+
+    if (lines[lines.length - 1] === '') {
+        //in case it wanted to count an empty line at the bottom
+        lines.pop();
+    }
+
+    return lines.length.toString();
+}
+
+function getWords(data) {
+    var regexSplit = /\s/;
+    var wordsInitial = data.split(regexSplit); //array
+
+    while (wordsInitial.indexOf('') !== -1) {
+        //with words, cut out any blanks enumerated
+        wordsInitial.splice(wordsInitial.indexOf(''), 1);
+    }
+    return wordsInitial.length.toString();
+}
+
+function getChars(data) {
+    return data.length.toString();
```

Comment:
I think this is actually wrong (I'm looking at [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length)):

> uses a single 16-bit code unit to represent the most common characters, but needs to use two code units for less commonly-used characters, **so it's possible for the value returned by length to not match the actual number of characters in the string.**

Is there a way to guarantee that two-codepoint-long characters only count as a single character each?

Reply:
I went back and forth on this a lot. It's been a while (in grad school time) so my mind has been elsewhere. 
There is a complication in determining the actual encoding and how that is counted. I also wanted to implement the "bytes" option -- anyways, I'll relook this as I was assuming the .length property was sufficient.

I think i'll read the file into a buffer and work on the buffer (instead of reading it in as UTF-8, as it currently stand), possibly first determining it's encoding, then moving on from there. That might require another library, we shall see.

Reply:
some more initial thoughts -- since i'm already reading this stream in as a utf-8, it should be in 'chunks' that made a 16-bit code unit into a character that .length property will count as one. Here is a demonstration, using the REPL:
```
> const buf5 = Buffer.from('tést');
undefined
> buf5
<Buffer 74 c3 a9 73 74>
> buf5.toString()
'tést'
> buf5.toString('utf-8')
'tést'
> buf5.toString('hex')
'74c3a97374'
> buf5.length
5
> buf5.toString().length
4
```

Reply:
Reference: https://nodejs.org/api/buffer.html#buffer_buffer

Reply:
also, based on this conversation, I think i'll implement the '--byte' option, which may be useful

Reply:
I think you need to test with a different character. Try '🎉' instead. 'é' can be encoded in a single utf-16 code unit (2 bytes).

> I think i'll implement the '--byte' option, which may be useful

Yeah, that's fine, as long as we get the correct bytes of the file, not the bytes of the JS string.

Also, I'm not sure what the correct behavior is when the file is utf-16 encoded. Unix `wc` doesn't seem to give correct answers.